query-engine/core/src/query_document/selection.rs

```diff
@@ -0,0 +1,310 @@
+use super::QueryValue;
+use itertools::Itertools;
+use std::{borrow::Cow, collections::BTreeMap};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct SelectionBuilder {
+    name: String,
+    alias: Option<String>,
+    arguments: Vec<(String, QueryValue)>,
+    nested_selections: Vec<Selection>,
+}
+
+impl SelectionBuilder {
+    pub fn alias(&mut self, value: impl Into<String>) -> &mut Self {
+        self.alias = Some(value.into());
+        self
+    }
+
+    pub fn arguments(&self) -> &[(String, QueryValue)] {
+        &self.arguments
+    }
+
+    pub fn set_arguments(&mut self, value: Vec<(String, QueryValue)>) -> &mut Self {
+        self.arguments = value;
+        self
+    }
+
+    pub fn push_argument(&mut self, key: impl Into<String>, arg: impl Into<QueryValue>) -> &mut Self {
+        self.arguments.push((key.into(), arg.into()));
+        self
+    }
+
+    pub fn nested_selections(&mut self, sels: Vec<Selection>) -> &mut Self {
+        self.nested_selections = sels;
+        self
+    }
+
+    pub fn push_nested_selection(&mut self, selection: Selection) -> &mut Self {
+        self.nested_selections.push(selection);
+        self
+    }
+
+    pub fn contains_nested_selection(&self, name: &str) -> bool {
+        self.nested_selections.iter().any(|sel| sel.name() == name)
```

Comment:
should we also compare the arguments here?

Reply:
apparently we only use it to compare selections to see if we are selecting the identifying fields,  and not to compare between 2 operations so it's ok